spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,669 @@
+        # Select healthy controls based on sex and/or age range
+        path_ref = os.path.join(__data_dir__, 'PAM50_normalized_metrics')
+        if sex or age:
+            if not os.path.isfile(get_absolute_path(os.path.join(path_ref, arguments.file_participants))):
+                raise FileNotFoundError('participants.tsv file must exists to select sex or age.')
+            else:
+                fname_partcipants = get_absolute_path(os.path.join(path_ref, arguments.file_participants))
+                list_HC = select_HC(fname_partcipants, sex, age)
+        else:
+            list_HC = None
+        if age:
+            age.sort()
+            if any(n < 0 for n in age):
+                parser.error('Age range needs to be positive, {} was specified'.format(age))
+            # Put age range in order
+            else:
+                age.sort()
+        # Get PAM50 slice thickness
+        fname_PAM50 = os.path.join(__data_dir__, 'PAM50', 'template', 'PAM50_t2.nii.gz')
+        img_pam50 = Image(fname_PAM50).change_orientation('RPI')
+        slice_thickness_PAM50 = get_slice_thickness(img_pam50)
+        # Fetch metrics of PAM50 template
+        df_metrics_PAM50 = pd.read_csv(fname_metrics_PAM50).astype({metric: float})
+        # Average metrics of healthy controls
+        df_avg_HC = average_hc(path_ref, metric, list_HC)
+
+        # Step 3. Determine compressed levels for both subject and PAM50 space
+        # --------------------------------------------------------------------
+        # Get slices correspondance in PAM50 space
+        compressed_levels_dict = get_slices_in_PAM50(compressed_levels_dict, df_metrics, df_metrics_PAM50)
+        z_range_below, z_range_above = get_slices_upper_lower_level_from_PAM50(compressed_levels_dict, df_metrics_PAM50, distance, extent, slice_thickness_PAM50)
+        # Get data from healthy control and average them
+    else:
+        # Get spinal cord centerline object to compute the distance
+        # Get max and min index of the segmentation with pmj
+        _, _, Z = (img_seg.data > NEAR_ZERO_THRESHOLD).nonzero()
+        min_z_index, max_z_index = min(Z), max(Z)
+        # Get the z index corresponding to the segmentation since the centerline only includes slices of the segmentation.
+        z_ref = np.array(range(min_z_index.astype(int), max_z_index.max().astype(int) + 1))
+        # Get centerline object
+        centerline = get_centerline_object(img_seg, verbose=verbose)
+        # Get healthy slices to average for level above and below
+        z_range_above, z_range_below = get_slices_upper_lower_level_from_centerline(centerline, distance, extent, slice_compressed, z_ref)
+
+    # Step 4. Compute MSCC metrics for each compressed level
+    # ------------------------------------------------------
+    # Loop through all compressed levels (compute one MSCC per compressed level)
+    for level in compressed_levels_dict.keys():
+        # Get metric of patient with compression
```

Comment:
Maybe, the `printv` from line 660 could be moved here to have `Level:` followed by `metric_a = ...` when using `-v 2`:

<img width="690" alt="image" src="https://user-images.githubusercontent.com/39456460/227304789-0516e07e-af2d-4203-98a6-50c1245a6026.png">


```suggestion
        printv('\nLevel: {}'.format(level), verbose=verbose, type='info')
        # Get metric of patient with compression
```

Reply:
Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4003/commits/2a587a74209da12d8dc638c022ea32474cc71293